src/test/java/uk/gov/companieshouse/service/AppealServiceTest.java

```diff
@@ -89,68 +92,89 @@ void testCreateAppeal_verify_createdBy_createdAt_setOnAppeal() {
 
-        assertThrows(ChipsServiceException.class, () -> appealService.saveAppeal(createAppeal(), TestData.Appeal.CreatedBy.id));
+        assertThrows(ChipsServiceException.class,
+            () -> appealService.saveAppeal(appeal, TestData.Appeal.CreatedBy.id));
 
-        verify(appealRepository).insert(createAppealEntity(null));
+        verify(appealRepository).insert(createAppealEntity(null, reasonEntity));
     }
 
     @Test
     void testGetAppealById_returnsAppeal() {
-
-        when(appealRepository.findById(any(String.class))).thenReturn(Optional.of(createAppealEntity(TestData.Appeal.id)));
-        when(appealMapper.map(any(AppealEntity.class))).thenReturn(createAppeal());
+        ReasonEntity reasonEntity = createReasonEntityWithOther();
+        Reason reason = createReasonWithOther();
+        when(appealRepository.findById(any(String.class))).thenReturn(
+            Optional.of(createAppealEntity(TestData.Appeal.id, reasonEntity)));
+        when(appealMapper.map(any(AppealEntity.class))).thenReturn(createAppeal(reason));
 
         Appeal appeal = appealService.getAppeal(TestData.Appeal.id).orElseThrow();
 
         assertEquals(TestData.Appeal.PenaltyIdentifier.companyNumber, appeal.getPenaltyIdentifier().getCompanyNumber());
-        assertEquals(TestData.Appeal.PenaltyIdentifier.penaltyReference, appeal.getPenaltyIdentifier().getPenaltyReference());
+        assertEquals(TestData.Appeal.PenaltyIdentifier.penaltyReference,
+            appeal.getPenaltyIdentifier().getPenaltyReference());
         assertEquals(TestData.Appeal.Reason.OtherReason.title, appeal.getReason().getOther().getTitle());
         assertEquals(TestData.Appeal.Reason.OtherReason.description, appeal.getReason().getOther().getDescription());
         assertEquals(2, appeal.getReason().getOther().getAttachments().size());
-        assertEquals(TestData.Appeal.Reason.Attachment.id, appeal.getReason().getOther().getAttachments().get(0).getId());
-        assertEquals(TestData.Appeal.Reason.Attachment.name, appeal.getReason().getOther().getAttachments().get(0).getName());
-        assertEquals(TestData.Appeal.Reason.Attachment.contentType, appeal.getReason().getOther().getAttachments().get(0).getContentType());
-        assertEquals(TestData.Appeal.Reason.Attachment.size, appeal.getReason().getOther().getAttachments().get(0).getSize());
+        assertEquals(TestData.Appeal.Reason.Attachment.id,
+            appeal.getReason().getOther().getAttachments().get(0).getId());
+        assertEquals(TestData.Appeal.Reason.Attachment.name,
+            appeal.getReason().getOther().getAttachments().get(0).getName());
+        assertEquals(TestData.Appeal.Reason.Attachment.contentType,
+            appeal.getReason().getOther().getAttachments().get(0).getContentType());
+        assertEquals(TestData.Appeal.Reason.Attachment.size,
+            appeal.getReason().getOther().getAttachments().get(0).getSize());
```

Comment:
These tests are hard to read. I recommend simplifying.

```
@DisplayName("Get appeal by ID")
@Test
void testGetAppealById() {
    Appeal appeal = createAppeal(createReasonWithOther());
    AppealEntity appealEntity = createAppealEntity(createReasonEntityWithOther());

    when(appealMapper.map(appeal).thenReturn(appealEntity);
    when(appealRepository.findById(TestData.Appeal.id)).thenReturn(Optional.of(appealEntity));

    assertEquals(appeal, appealService.getAppeal(TestData.Appeal.id));
}
```

Reply:
Test constants were used to simplify the unit tests. Refactoring to remove unnecessary assertions has been added into a unit test refactoring task. This will be resolved separately to this PR.